rpc/gnmi/gnmi.proto

```diff
@@ -87,8 +87,27 @@ message Notification {
 // Value pair.
 // Reference: gNMI Specification Section 2.1
 message Update {
-  Path path = 1;    // The path (key) for the update.
-  Value value = 2;  // The value (value) for the update.
+  Path path = 1;                      // The path (key) for the update.
+  Value value = 2 [deprecated=true];  // The value (value) for the update.
+  // One of the fields within the val oneof is populated with the value
+  // of the update. The type of the value being included in the Update
+  // determines which field should be populated. In the case that the
+  // encoding is a particular form of the base protobuf type, a specific
+  // field is used to store the value (e.g., json_val).
+  oneof val {
```

Comment:
One advantage of having an explicit message type for the value containing the oneof is that it can be referenced in other places. I am leaning in favor of having a single message type with the oneof that can be used both for the value and the leaflist with comments clearly identifying which elements are not eligible to be leaflist members.